http/src/handler.rs

```diff
@@ -189,7 +194,19 @@ impl<M: Metadata, S: Middleware<M>> Future for RpcHandler<M, S> {
 				RpcPollState::Ready(self.read_headers(request, continue_on_invalid_cors))
 			},
 			RpcHandlerState::ReadingBody { body, request, metadata, uri, } => {
-				self.process_body(body, request, uri, metadata)?
+				match self.process_body(body, request, uri, metadata) {
+					Err(BodyError::Utf8(ref e)) => {
+						let mesg = format!("utf-8 encoding error at byte {} in request body", e.valid_up_to());
```

Comment:
Perhaps `415 Unsupported Media Type`? But I'm fine with 400 as well.

> The 415 (Unsupported Media Type) status code indicates that the
origin server is refusing to service the request because the payload
is in a format not supported by this method on the target resource.
The format problem might be due to the request's indicated
Content-Type or Content-Encoding, or as a result of inspecting the
data directly.

Reply:
Since we already verified the content-type is `application/json` or `application/json; charset=utf-8` and [RFC8259](https://tools.ietf.org/html/rfc8259) states:

>  JSON text exchanged between systems that are not part of a closed    ecosystem MUST be encoded using UTF-8

I think a 400 is appropriate.